### ISSUE TITLE
[cluster-test] [Node launcher 1-3/n] Separate nodes and instances

### ISSUE THREAD

**Issue by andll** [libra/libra#5063]
There going to be more issues in the thread. Goal of all of this is to change how we launch libra nodes in cluster test to allow higher degree of flexibility when writing tests and unblocking some scenarios that are not currently possible.

Part 1-3 of this journey will be removing usage of libra-init from validator/full node specs.

### Seed IP bash magic
Currently when Instance runs, it initialize it's internal state by [running libra-init in the init container](https://github.com/libra/libra/blob/master/testsuite/cluster-test/src/cluster_swarm/validator_spec_template.yaml#L19).

This code contains bit of bash magic, that limits what we can do and hard to maintain.

We want to reduce amount of bash magic there :)

Specifically, init container does following:

- Configures fluentbit. *This part we can actually leave as is for now.*
- Removes data dir when `delete_data` is set
- Populates `safety_rules_addr` from kubernetes
- Populates `seed_peer_ip` from kuberenetes

The last 3 pieces is what we want to replace. Instead of searching for those values in Kubernetes, we want to pre-allocate nodes and populate those values right away in the cluster test.

Instead of figuring out IP addresses inside the container, we want to pre-allocate all the nodes in `ClusterBuilder`, and include ip addresses to the containers configuration before launching them.

The way this can be done:

* Make `ClusterSwarmKube::allocate_node` public. Currently node allocation is done internally in ClusterSwarmKube, but we want to change it.
* In `ClusterBuilder`, we should pre-allocate nodes for all containers using public method above.
* Add appropriate fields(safety_rules_addr/seed_peer_ip) to the ValidatorConfig/FullNodeConfig. Note that seed_peer_ip has different meaning for validators and full nodes - for validators this is ip of pod 0 for validator, while for full node this is IP address of appropriate validator
* In ClusterSwarmKube::validator_spec/fullnode_spec we should use those parameters in formatting spec template
* In the validator|fullnode_spec_template files, we need to remove kubectl magic and usage of `/opt/libra/data/seed_peer_ip`, etc files, and use supplied parameters instead.

### Delete data
After addressing seed IP problem, we need to look at the delete_data(in a separate PR):

* delete_data parameter should be removed from template and code, like `Instance::start` and `spawn_new_instance`. Default should be false(this is also most common I believe)
* Instead of supplying it as a parameter, we want to introduce method on the Instance, such as `delete_data`. In the existing code, where we previously passed delete_data=true, we instead want to call Instance::clean_data before running Instance::start
* `delete_data` on the instance can be a shortcut to Instance::util_cmd and call `rm -rf`

### Fluentbit configuration

Last piece of init container in validator/full node is initializing fluentbit.

This can be done by adding method such as `ClusterSwarmKube::put_file(&self, node: KubeNode, file: &[u8])`. This can be done, for example, by running [kubectl cp](https://medium.com/@nnilesh7756/copy-directories-and-files-to-and-from-kubernetes-container-pod-19612fa74660) in cluster test. Example of running kubectl see `Instance::exec`.
